command/ca/provisioner/caConfigClient.go

```diff
@@ -0,0 +1,202 @@
+package provisioner
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/authority/config"
+	"github.com/smallstep/certificates/authority/provisioner"
+	"github.com/smallstep/certificates/ca"
+	"go.step.sm/cli-utils/ui"
+	"go.step.sm/linkedca"
+)
+
+// nodb implements the certificates/Admiclient interface with noops.
+type nodb struct{}
+
+func newNoDB() *nodb {
+	return &nodb{}
+}
+
+func (n *nodb) CreateProvisioner(ctx context.Context, prov *linkedca.Provisioner) error {
+	return nil
+}
+
+func (n *nodb) GetProvisioner(ctx context.Context, id string) (*linkedca.Provisioner, error) {
+	return nil, nil
+}
+
+func (n *nodb) GetProvisioners(ctx context.Context) ([]*linkedca.Provisioner, error) {
+	return nil, nil
+}
+
+func (n *nodb) UpdateProvisioner(ctx context.Context, prov *linkedca.Provisioner) error {
+	return nil
+}
+
+func (n *nodb) DeleteProvisioner(ctx context.Context, id string) error {
+	return nil
+}
+
+func (n *nodb) CreateAdmin(ctx context.Context, admin *linkedca.Admin) error {
+	return nil
+}
+
+func (n *nodb) GetAdmin(ctx context.Context, id string) (*linkedca.Admin, error) {
+	return nil, nil
+}
+
+func (n *nodb) GetAdmins(ctx context.Context) ([]*linkedca.Admin, error) {
+	return nil, nil
+}
+
+func (n *nodb) UpdateAdmin(ctx context.Context, prov *linkedca.Admin) error {
+	return nil
+}
+
+func (n *nodb) DeleteAdmin(ctx context.Context, id string) error {
+	return nil
+}
+
+type caConfigClient struct {
+	configFile string
+	ctx        context.Context
+	auth       *authority.Authority
+}
+
+func newCaConfigClient(ctx context.Context, cfg *config.Config, cfgFile string) (*caConfigClient, error) {
+	a, err := authority.New(cfg, authority.WithAdminDB(newNoDB()), authority.WithSkipInit())
+	if err != nil {
+		return nil, errors.Wrapf(err, "error loading authority")
+	}
+	client := &caConfigClient{
+		configFile: cfgFile,
+		ctx:        ctx,
+		auth:       a,
+	}
+	if err := client.auth.ReloadAdminResources(ctx); err != nil {
+		return nil, errors.Wrapf(err, "error loading provisioners from config")
+	}
```

Comment:
Will this fail if we want to change a bad OIDC configuration present in the ca.json?

Reply:
Good catch! I think we have two general options here:

1) I can build the provisioner collection my self and then set it on the Authority using something like With ProvisionerCollection - OR -
2) I can lazy load the provisioner - which could have the benefit of fixing the bug where the CA becomes unusable if, for some reason, the OIDC configuration changes and breaks the currently stored one.